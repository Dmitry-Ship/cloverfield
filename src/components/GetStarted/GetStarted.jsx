import React, { PropTypes } from 'react';
import Button from '../basic/Button';

import { buttonLeft, buttonRight } from './GetStarted.scss';

const GetStarted = props => {
  const style = {
    // display: 'flex',
    // flexWrap: 'nowrap',
    // justifyContent: 'center',
    // alignItems: 'center',
    // width: '100%',
    position: 'relative',
  };
  return (
    <div style={style} >
      <Button className={buttonLeft} onClick={props.openSignUpModal} >Get started for free</Button>
      <Button className={buttonRight} kind="secondary" onClick={props.openLoginModal} >I already have an account</Button>
    </div>
  );
};

GetStarted.propTypes = {
  openLoginModal: PropTypes.func.isRequired,
  openSignUpModal: PropTypes.func.isRequired,
};

export default GetStarted;
