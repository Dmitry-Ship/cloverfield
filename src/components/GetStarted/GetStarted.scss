.buttonLeft {
  position: absolute;
  right: 55%;
}

.buttonRight {
  position: absolute;
  left: 55%;
}