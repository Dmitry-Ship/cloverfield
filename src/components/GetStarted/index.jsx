export { default } from './GetStarted.jsx';
