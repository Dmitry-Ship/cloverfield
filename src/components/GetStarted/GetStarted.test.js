import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import GetStarted from './GetStarted';

describe('<GetStarted />', () => {
  const openSignUpModal = jest.fn();
  const openLoginModal = jest.fn();

  it('should render correctly', () => {
    const tree = renderer.create(
      <GetStarted openSignUpModal={openSignUpModal} openLoginModal={openLoginModal} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call openSignUpModal', () => {
    const wrapper = shallow(
      <GetStarted openSignUpModal={openSignUpModal} openLoginModal={openLoginModal} />
    );

    wrapper.find('Button').first().simulate('click');

    expect(openSignUpModal.mock.calls.length).toBe(1);
  });

  it('should call openLoginModal', () => {
    const wrapper = shallow(
      <GetStarted openSignUpModal={openSignUpModal} openLoginModal={openLoginModal} />
    );

    wrapper.find('Button').at(1).simulate('click');

    expect(openLoginModal.mock.calls.length).toBe(1);
  });
});
