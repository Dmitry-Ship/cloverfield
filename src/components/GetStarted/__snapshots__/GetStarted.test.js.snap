// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetStarted /> should render correctly 1`] = `
<div
  className="wrapper"
>
  <button
    className="primary normal buttonLeft"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Get started for free
  </button>
  <button
    className="secondary normal buttonRight"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    I already have an account
  </button>
</div>
`;
